compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/IrInlineCodegen.kt

```diff
@@ -194,6 +195,41 @@ class IrInlineCodegen(
             ::IrDefaultLambda
         )
     }
+
+    override fun inlineCall(
+        nodeAndSmap: SMAPAndMethodNode,
+        inlineDefaultLambda: Boolean,
+        isCallOfFunctionInCorrespondingDefaultDispatch: Boolean,
+    ): InlineResult {
+        if (!isCallOfFunctionInCorrespondingDefaultDispatch)
+            return super.inlineCall(nodeAndSmap, inlineDefaultLambda, isCallOfFunctionInCorrespondingDefaultDispatch)
+
+        // The call to a function in its corresponding $default handler is inlined verbatim in the JVM BE, so for compatibility
+        // we have to do the same here. The code below is mostly copied from InlineCodegenForDefaultBody.kt.
+        val childSourceMapper = SourceMapCopier(sourceCompiler.lazySourceMapper, nodeAndSmap.classSMAP)
+
+        val node = nodeAndSmap.node
+        val transformedMethod = MethodNode(
+            node.access,
+            node.name,
+            node.desc,
+            node.signature,
+            node.exceptions.toTypedArray()
+        )
+
+        val argsSize =
+            (Type.getArgumentsAndReturnSizes(jvmSignature.asmMethod.descriptor) ushr 2) - if (function.isStatic) 1 else 0
+        node.accept(object : InlineAdapter(transformedMethod, 0, childSourceMapper) {
+            override fun visitLocalVariable(name: String, desc: String, signature: String?, start: Label, end: Label, index: Int) {
+                // We only copy LVT entries for local variables, since we already generated entries for the method parameters,
+                if (index >= argsSize) super.visitLocalVariable(name, desc, signature, start, end, index)
+            }
+        })
+
+        transformedMethod.accept(MethodBodyVisitor(codegen.visitor))
```

Comment:
Looks equivalent to just passing `MethodBodyVisitor` to `InlineAdapter`

Reply:
Good call, I didn't even look into the method visitors. This simplified the code a bit, thanks. :)